### PR TITLE
Cardinality Issues with Extension-DiagnosticReport.media.link

### PR DESCRIPTION
**Pull request by KazeemHamzat**
Changing the cardinality of Extension-DiagnosticReport.media.link to 0 to many (0...*) instead of the current 1 to many(1..*)

### CODE REVIEW

**Review by vickyjaiswal0308 — APPROVED** [review]

**Review by ryma2fhir — CHANGES_REQUESTED** [review]
@KazeemHamzat @vickyjaiswal0308 
I'm not quite getting this one. I see we already have DiagnosticReport.media which contains .link 1..1 and .comment 0..1

I do see that the link reference has changed in R5 from `media` to `DocumentReference`, which is what you want adding.

I'm happy to keep the media 0..*, and allow both `media` and `DocumenentReference`, but it should be one or another (you either follow R4 or R5 not both, so we'd need a FHIRpath to stop both being used at the same time. 

The `comment` can only have a maximum of 1 but by adding this to the extension you have allowed a maximum of 2.  

So in my mind, the extension is media.link only (no comment element) and the slice name is mediaLinkR5, and can only be allowed on the `DiagnosticReport.media` and only one type of media.link allowed (R4 xor R5 reference)

**Review by vickyjaiswal0308 — COMMENTED** [review]
Change the URL to a UK-specific one to clarify that it is a backport:
https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-DiagnosticReportMediaLink

**Review by vickyjaiswal0308 — COMMENTED** [review]
The min="1" on Extension.value[x] ensures that the value is mandatory. If mandatory usage is not required, set min="0".

**Comment by KazeemHamzat** [comment]
> Change the URL to a UK-specific one to clarify that it is a backport: https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-DiagnosticReportMediaLink

This is backported element from FHIR R5, so its URL will be that of R5

File: structuredefinitions/UKCore-DiagnosticReport.xml [hunk 1]
```diff
@@ -149,6 +131,25 @@
...
       <mustSupport value="true" />
     </element>
+    <element id="DiagnosticReport.media">
+      <path value="DiagnosticReport.media" />
+      <constraint>
+        <key value="ukcore-diag-lab-002" />
+        <severity value="error" />
+        <human
+          value="Only one of DiagnosticReport.media.link or DiagnosticReport.media:mediaLinkR5 can be used at a time." />
+        <expression value="media.where(sliceName.exists()).count() = 0 or media.link.exists() = false" />
```
Comment (ryma2fhir) [inline structuredefinitions/UKCore-DiagnosticReport.xml:141]: There isn't a slice anymore. Can you follow https://github.com/NHSDigital/FHIR-R4-UKCORE-STAGING-MAIN/blob/fad692ab52970801afb9bbbf90e260fe8b57fdb1/structuredefinitions/UKCore-Specimen.xml#L35C9-L36C199:

```<human value="There SHALL be only one reference between `collection.collector` and the extension collectionCollectorR5`" />```
```<expression value="collection.collector.reference.empty() or collection.collector.extension('http://hl7.org/fhir/5.0/StructureDefinition/extension-Specimen.collection.collector').empty()" />```

**Review by ryma2fhir — CHANGES_REQUESTED** [review]

**Comment by ryma2fhir** [comment]
Sorry @KazeemHamzat I've got this totally wrong. I've just realised `media.link` is `1..1` so we can't have the R5 version without the R4 version also being added.

You had it correct the first time (I think):
`mediaR5 extension` for the media R5 version. This will include both `media.comment` and `media.link(DocumentReference)`
Then FHIRPath to say one or the other exists.

The following scenarios should be tested:
No media or mediaR5 = PASS
Media but no mediaR5 = PASS
no media but a mediaR5 = PASS
media and mediaR5 = FAIL


Apologies for that, I have no idea what I was thinking yesterday

**Review by ryma2fhir — CHANGES_REQUESTED** [review]
I see we still have the extension to `media.link`, this should now be `media` 

I'm also thinking the `media.comment` and `media.link` wont reference the R5, so it might be that the fixedUri's need to be the R5 extension url? I'm not entirely sure on this one, could you ask on [chat.fhir](https://chat.fhir.org/), you will need to supply them with the two options.

File: structuredefinitions/UKCore-DiagnosticReport.xml [hunk 2]
```diff
@@ -27,6 +27,15 @@
   <baseDefinition value="http://hl7.org/fhir/StructureDefinition/DiagnosticReport" />
   <derivation value="constraint" />
   <differential>
+    <element id="DiagnosticReport">
+      <path value="DiagnosticReport" />
+      <constraint>
+        <key value="ukcore-diag-lab-002" />
+        <severity value="error" />
+        <human value="Only one of DiagnosticReport.media.link or DiagnosticReport.media.extension:mediaLinkR5 can be populated at a time." />
+        <expression value="media.empty() or extension('http://hl7.org/fhir/5.0/StructureDefinition/extension-DiagnosticReport.media.link').empty()" />
```
Comment (ryma2fhir) [inline structuredefinitions/UKCore-DiagnosticReport.xml:36]: The word 'link' needs removing from both lines, so should be:

<human value="Only one of DiagnosticReport.media or DiagnosticReport.media.extension:mediaR5 can be populated at a time." />
        <expression value="media.empty() or extension('http://hl7.org/fhir/5.0/StructureDefinition/extension-DiagnosticReport.media').empty()" />

**Review by ryma2fhir — CHANGES_REQUESTED** [review]

**Review by ryma2fhir — DISMISSED** [review]

**Review by vickyjaiswal0308 — DISMISSED** [review]
Seems fine to me

**Comment by ryma2fhir** [comment]
My GPG key didnt work on signing commit - I'm happy it was me so pushing to develop

**Pull request merged**